scripts/publish.ts

```diff
@@ -0,0 +1,104 @@
+import { $ } from "execa";
+import mri from "mri";
+import semverPrerelease from "semver/functions/prerelease";
+import semverValid from "semver/functions/valid";
+
+publish();
+
+type Options = { defaultBranchName?: string };
+
+async function publish(options?: Options): Promise<void> {
+  const defaultBranchName = options?.defaultBranchName ?? "main";
+
+  const args = mri<{ "dry-run": boolean }>(process.argv.slice(2), {
+    alias: { n: "dry-run" },
+    boolean: ["dry-run"],
+    default: { "dry-run": false },
+  });
+
+  const mode = args._[0];
+  if (mode !== "stable" && mode !== "unstable") {
+    console.error("The first argument must be either `stable` or `unstable`.");
+    process.exit(1);
+  }
+  const specifiers = args._.slice(1);
+  if (mode === "stable" && specifiers.length === 0) {
+    console.error(
+      "You must specify at least one package version (using the `<name>@<version>` format) to create a stable release.",
+    );
+    process.exit(1);
+  } else if (mode === "unstable" && specifiers.length > 0) {
+    console.error(
+      "You can't specify any package version when creating an unstable release.",
```

Comment:
Small phrasing change suggestion: "Versions cannot be specified when creating an unstable release."